File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataAvatarModel.java
package com.mcreater.amclcore.api.mihoyo.genshin.model;

import com.google.gson.annotations.SerializedName;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class GenshinGachaDataAvatarModel {
    private int id;
    private String name;
    private String icon;
    private int element;
    private int level;
    @SerializedName("weapon_cat_id")
    private int weaponCatId;
    @SerializedName("set_list")
    private List<Object> setList;
    @SerializedName("wiki_url")
    private String wikiUrl;
    @SerializedName("first_attr")
    private List<Object> firstAttr;
    @SerializedName("secondary_attr")
    private List<Object> secondaryAttr;
    @SerializedName("secondary_attr_name")
    private List<Object> secondaryAttrName;
    @SerializedName("head_icon")
    private String headIcon;
    @SerializedName("pc_icon")
    private String pcIcon;
}

File: core/src/main/java/com/mcreater/amclcore/api/mihoyo/genshin/model/GenshinGachaDataAvatarModel.kt
package com.mcreater.amclcore.api.mihoyo.genshin.model

import com.google.gson.annotations.SerializedName

data class GenshinGachaDataAvatarModel(
    var id: Int = 0,
    var name: String? = null,
    var icon: String? = null,
    var element: Int = 0,
    var level: Int = 0,
    @SerializedName("weapon_cat_id") var weaponCatId: Int = 0,
    @SerializedName("set_list") var setList: List<Any>? = null,
    @SerializedName("wiki_url") var wikiUrl: String? = null,
    @SerializedName("first_attr") var firstAttr: List<Any>? = null,
    @SerializedName("secondary_attr") var secondaryAttr: List<Any>? = null,
    @SerializedName("secondary_attr_name") var secondaryAttrName: List<Any>? = null,
    @SerializedName("head_icon") var headIcon: String? = null,
    @SerializedName("pc_icon") var pcIcon: String? = null
)